### ISSUE TITLE
Qualified keywords can't be used as path-parameters

### ISSUE THREAD

**Issue by ikitommi** [metosin/reitit#199]
non-qualified keywords work:

```clj
; wanted :id
["/user/:id/orders"]
; => :id
```

fully qualified keywords are split at slash:

```clj
; wanted :my.domain/id
["/user/:my.domain/id/orders"]
; => :my.domain
```

to support fully qualified keywords, we need new path-parameter syntax, maybe something like:

```clj
; wanted :my.domain/id
["/user/:my.domain\\/id/orders"]
["/user/{:my.domain/id}/orders"]
; => :my.domain/id
```

the latter, e.g. allowing the path-parameters to be wrapped into curly brackets, could still only support full-segment parameters (like now), but pave the way to support partial segments too, e.g.

```clj
["/files/{:file}.json"]
```

(this would require `segment-router` to be rewritten)

we still want the syntax to be simple = strings.

**Comment by miikka** [comment]
I wonder if it would make architectural sense to have bidi-style EDN syntax/AST for the routes that would be consumed by the routers, and a string syntax that would be converted to the AST.

**Comment by ikitommi** [comment]
Internally, it could be like that (and there is the `r/route-info` to get ~that data out). Having an alternative syntax for users, today I would say no, as it would make things more complex. Router hooks (transform, expand, compile), `Match` and functions like `r/routes` and `r/compiled-routes` would also be effected.

**Comment by sebn** [comment]
Would it make sense for coercion to be defined once per qualified keyword, i.e. `:my.app.person/id` would always be coerced in the same way, whatever the route? Or am I missing something?

**Comment by ikitommi** [comment]
This is fixed in #215. I think we should support both formats for some time..

**Comment by ikitommi** [comment]
the new Trie solved this (merged in master):

```clj
[["/users/{user/id}"]
 ["/files/file-{file/number}.pdf"]]
```
